src/sources/docker.rs

```diff
@@ -379,256 +462,121 @@ impl DockerSource {
-                            }
-                        }
-                        Err(error) => error!(source="docker events",%error),
-                        // Stream has ended
-                        Ok(Async::Ready(None)) => {
-                            // TODO: this could be fixed, but should be tryed with some timeoff and exponential backoff
-                            error!(message = "docker event stream has ended unexpectedly");
-                            info!(message = "Shuting down docker source");
-                            return Err(());
-                        }
-                    }
-                }
-                Err(()) => error!(message = "Error in docker source main stream"),
-                // For some strange reason stream has ended.
-                // It should never reach this point. But if it does,
-                // something has gone terrible wrong, and this system is probably
-                // in invalid state.
-                Ok(Async::Ready(None)) => {
-                    error!(message = "docker source main stream has ended unexpectedly");
-                    info!(message = "Shuting down docker source");
-                    return Err(());
-                }
-            }
-        }
-    }
-}
-
 /// Used to construct and start event stream futures
 #[derive(Clone)]
 struct EventStreamBuilder {
     core: Arc<DockerSourceCore>,
     /// Event stream futures send events through this
-    out: Sender<Event>,
+    out: mpsc01::Sender<Event>,
     /// End through which event stream futures send ContainerLogInfo to main future
-    main_send: UnboundedSender<ContainerLogInfo>,
+    main_send: mpsc::UnboundedSender<ContainerLogInfo>,
     /// Self and event streams will end on this.
     shutdown: ShutdownSignal,
 }
 
 impl EventStreamBuilder {
-    fn new(
-        core: DockerSourceCore,
-        out: Sender<Event>,
-        main_send: UnboundedSender<ContainerLogInfo>,
-        shutdown: ShutdownSignal,
-    ) -> Self {
-        EventStreamBuilder {
-            core: Arc::new(core),
-            out,
-            main_send,
-            shutdown,
-        }
-    }
-
     /// Constructs and runs event stream until shutdown.
     fn start(&self, id: ContainerId) -> ContainerState {
-        let metadata_fetch = self
-            .core
-            .docker
-            .containers()
-            .get(id.as_str())
-            .inspect()
-            .map_err(|error| error!(message="Fetching container details failed",%error))
-            .and_then(|details| {
-                ContainerMetadata::from_details(&details)
-                    .map_err(|error| error!(message="Metadata extraction failed",%error))
-            });
-
         let this = self.clone();
-        let task = metadata_fetch.and_then(move |metadata| {
-            this.start_event_stream(ContainerLogInfo::new(id, metadata, this.core.now_timestamp))
+        tokio::spawn(async move {
+            Arc::clone(&this.core)
```

Comment:
We can just use `this.core.clone()`

Reply:
Yes we can, but I think `Arc::clone` is more clear.

Reply:
Is the information that we are cloning `Arc` and doing it by borrowing `this.core` relevant here? If not, `this.core.clone()` has less noise.